utils/vector.hpp

```diff
@@ -0,0 +1,366 @@
+#pragma once
+
+namespace nstd
+{
+template<class T>
+class Vector
+{
+public:
+    using uint = unsigned int;
+    using reference = T&;
+    using iterator  = T*;
+    using const_reference = const T&;
+    using const_iterator  = const T*;
+
+
+    /* Constructors */
+    Vector();
+    Vector(const Vector<T>&);
+    Vector(Vector<T>&&);
+    explicit Vector(uint);
+
+
+    /* Destructor */
+    ~Vector();
+
+
+    /* Capacity */
+    uint getSize() const;
+    uint getCapacity() const;
+    void resize(uint);
+    void resize(uint, const_reference);
```

Comment:
IMO this is kinda weird, naming the parameter something ambiguous like `const_reference`. I would prefer only using these aliases for the iterators if possible